### PR TITLE
Add language version to search

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
This PR adds the descriptor type version to search so it can be displayed as a Language Versions facet in the UI. The Language Versions facet allows users to filter for tools/workflows that contain files with the workflow language versions.

View PR https://github.com/dockstore/dockstore-ui2/pull/1673 for the UI changes.

Note that our legacy tools can have more than one descriptor language (CWL & WDL). This PR only adds the language versions to search for legacy tools that have one descriptor language to simplify things and because we plan to get rid of WDL tools and deprecate the legacy tool registration eventually ([docs](https://docs.dockstore.org/en/stable/advanced-topics/dockstore-tools-overhaul.html#future-of-dockstore-tools)). From what I can tell, prod has < 10 legacy tools that are both CWL and WDL.

**Review Instructions**
View the review instructions in the UI PR https://github.com/dockstore/dockstore-ui2/pull/1673

**Issue**
[SEAB-5052](https://ucsc-cgl.atlassian.net/browse/SEAB-5052)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead. 

[SEAB-5052]: https://ucsc-cgl.atlassian.net/browse/SEAB-5052?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java [hunk 1]
```diff
@@ -482,6 +484,27 @@ private static Set<String> getVerifiedPlatforms(Set<? extends Version> workflowV
...
+            // Only set descriptor type versions if there's one descriptor type otherwise we can't tell which version belongs to which type without looking at the source files
+            language = ((Tool)entry).getDescriptorType().get(0);
+        } else if (entry instanceof BioWorkflow || entry instanceof AppTool) {
+            language = ((Workflow)entry).getDescriptorType().toString();
+        } else {
+            return List.of();
+        }
+
+        // Get a list of unique descriptor type versions with the descriptor type prepended. Ex: 'WDL 1.0'
+        List<String> descriptorTypeVersions = workflowVersions.stream()
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java:499]: It would be good if we didn't have to load all the versions to calculate this, and calculate this with JQL instead. But we already load all the versions on lines 316 and 317, so it's only worth it if those get fixed as well.

Perhaps a followup ticket and/or todo?

**Review by coverbeck — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java [hunk 2]
```diff
@@ -482,6 +484,27 @@ private static Set<String> getVerifiedPlatforms(Set<? extends Version> workflowV
         return platforms;
     }
 
+    private static List<String> getDistinctDescriptorTypeVersions(Entry entry, Set<? extends Version> workflowVersions) {
+        String language;
+        if (entry instanceof Tool && ((Tool)entry).getDescriptorType().size() == 1) {
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java:489]: There's an interesting Java 16 syntax pointed out in Sonar cloud. 
May be useful to try out and may be worth running an intellj refactoring on the whole codebase if supported (can be separate PR)

**Review by denis-yuen — APPROVED** [review]
a few minor sonar cloud suggestions may be worth looking at if going back in, otherwise can be a dedicated (mostly automated) refactoring PR

**Review by svonworl — APPROVED** [review]
